railties/lib/rails/engine.rb

```diff
@@ -559,13 +561,12 @@ def config
     # Blog::Engine.load_seed
     def load_seed
       seed_file = paths["db/seeds.rb"].existent.first
-      return unless seed_file
+      executor.wrap { load(seed_file) } if seed_file
+    end
 
-      if config.try(:active_job)&.queue_adapter == :async
-        with_inline_jobs { load(seed_file) }
-      else
-        load(seed_file)
-      end
+    initializer :set_executor, before: :load_environment_hook do |app|
+      @executor = app.executor
+      @reloader = app.reloader
     end
```

Comment:
Is this a bad idea?  An alternative would be to add a `load_seed` callback and something like the following:

```ruby
initializer :wrap_load_seed_with_executor do |app|
  self.class.set_callback(:load_seed, :around) do |_, loading|
    app.executor.wrap(&loading)
  end
end
```

Reply:
I don't think engines should have their own executors and reloaders. I think we should go with a different implementation.